test/src/singletons.test.js

```diff
@@ -0,0 +1,28 @@
+require('../../src/setup');
+const singletons = require('../../src/singletons');
+const { ERC1820_REGISTRY_BYTECODE } = require('../../src/data');
+
+contract('singletons', function ([funder]) {
+  describe('ERC1820Registry', function () {
+    before(async function () {
+      this.registry = await singletons.ERC1820Registry(funder);
+    });
+
+    it('returns a truffle-contract instance', function () {
+      this.registry.constructor.name.should.equal('TruffleContract');
```

Comment:
I think using the `instanceof` operator would be a better test... But JavaScript is weird anyway. Did you choose this way for any reason in particular?

Reply:
`truffle-contract` doesn't seem to expose the underlying object - only a function to create those. So I'm not sure how we'd go about using `instanceof`, other than creating a new contract and comparing types (which requires an abi, binary, etc.).